### ISSUE TITLE
Slider doesn't update if value is changed during watch

### ISSUE THREAD

**Issue by jfneis** [NightCatSama/vue-slider-component#474]
We have a situation where, on a value change, we call an API and validate if the selected value is valid. If not, we will set the value to the closest one.

The problem is that slider doesn't visually update the value after it's set inside the method. If we get "value" it has the correct value, but visually slider stays where it was last dragged.

Also, if after the first change the "value" is changed by another component, slider doesn't move anymore. It simply stays frozen on the last dragged position (drag keeps working "normally", except it doesn't go to the 2nd set value).

I created an example on fiddle (https://jsfiddle.net/np7mywu8/23/) where:
- Value starts on 5 for both Slider and an Input
- If the value is changed, a watch sets the new value to 7: input gets it, Slider doesn't
- There is a button that increases value. On first use Slider works fine (from 5 to 7), after that, if dragged, it stops moving back to 7.

**Comment by NightCatSama** [comment]
Because dragging updates the value in real time, you can set `lazy="true"` to let the value update lazily.

Or do not use watch to monitor the value, you can change the judgment value through `@change` or `@drag-end` event.

**Comment by jfneis** [comment]
Setting lazy = true or not doesn't affect the problem, same thing happens.

I tried different combinations of using watch/@change/@drag-end but any of them works as expected. At the end of the day the slider, when dragged, doesn't respect the value set during the events or the watch (the input does).

I couldn't find documentation about using @change event. Am I missing something? Updated example: https://jsfiddle.net/4p5zu9bs/1/
